workflow/executor/pns/pns.go

```diff
@@ -141,50 +141,7 @@ func (p *PNSExecutor) Wait(ctx context.Context, containerNames, sidecarNames []s
 	}
 	p.rootFS = rootFS
 
-	/*
-		What is a "short running container" and "late starting container"?:
-
-		Short answer: any container that exits in <5s
-
-		Long answer:
-
-		Some containers are short running and we cannot determine their PIDs because they exit too quickly.
-		This loop allows 5s for `pollRootProcesses` find PIDs, so we define any container that exits <5s as short running
-
-		Unfortunately, we cannot assume that a container that did not appeared within the 5s has completed.
-		They may still be in `ContainerCreating` state - i.e. late starting.
-	*/
-	for i := 0; !p.haveContainerPIDs(containerNames) && i < 5; i++ {
-		time.Sleep(1 * time.Second)
-	}
-
-	if !p.haveContainerPIDs(containerNames) {
-		log.Info("container PIDs still unknown (maybe short running container, or late starting)")
-		return p.K8sAPIExecutor.Wait(ctx, containerNames, sidecarNames)
-	}
-
-OUTER:
-	for _, containerName := range containerNames {
```

Comment:
Very rarely, containers can successfully complete, but the Kubernetes API does not yet know about it, so `kubectl logs` will then fail, resulting in the wait container failing.

I don't see any easy way around this, so this reverts us to how v2.12.